demo_net/ZeroDowntimeDeployment/Middlewares/GracefulShutdownMiddleware.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Runtime.Loader;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace ZeroDowntimeDeployment.Middlewares
+{
+    public class GracefulShutdownMiddleware
```

Comment:
Could we add more logging and add 5 seconds sleep before we do the graceful shutdown. With 5 seconds sleep, we can test whether the code below works.